deflate.c

```diff
@@ -309,7 +308,11 @@ int ZEXPORT deflateInit2_(strm, level, method, windowBits, memLevel, strategy,
 
     s->wrap = wrap;
     s->gzhead = Z_NULL;
-    s->w_bits = (uInt)windowBits;
+#if MIN_LOOKAHEAD < 256
+    s->w_bits = ((uInt)windowBits >= 8) ? (uInt)windowBits : 8;
+#else
+    s->w_bits = ((uInt)windowBits >= 9) ? (uInt)windowBits : 9;
+#endif
     s->w_size = 1 << s->w_bits;
```

Comment:
As far as i can understand, a 256 byte window is not supported at all #171. Therefore, it is more correct to write like this:
```diff
-    s->w_bits = (uInt)windowBits;
+    s->w_bits = ((uInt)windowBits >= 9) ? (uInt)windowBits : 9;
```